packages/ckeditor5-engine/src/view/placeholder.ts

```diff
@@ -20,6 +20,8 @@ import type { ObservableChangeEvent } from '@ckeditor/ckeditor5-utils';
 // Each document stores information about its placeholder elements and check functions.
 const documentPlaceholders = new WeakMap<Document, Map<Element, PlaceholderConfig>>();
 
+let hasDisplayedWarning = false;
```

Comment:
The name could be a bit more specific in a sense to tell what kind of warning is it.